### PR TITLE
Revert "ext_auth: add option to measure timeout when check request is created. (#12778)"

### PR DESCRIPTION
**Pull request by yuval-k**
Commit Message:
Revert "ext_auth: add option to measure timeout when check request is created. (#12778)"

### CODE REVIEW

**Review by mattklein123 — DISMISSED** [review]
Thanks!

**Comment by yuval-k** [comment]
/retest

I think a CI issue, as it failed building `//test/common/secret:sds_api_test`

**Comment by mattklein123** [comment]
Looks like legit CI issue.

/wait

**Comment by yuval-k** [comment]
weird, as it builds fine locally. will try re-building with the exact clang version.

**Comment by yuval-k** [comment]
ok, merging latest master reproduced the failure; fixing

**Review by mattklein123 — DISMISSED** [review]

File: test/common/secret/sds_api_test.cc [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 #include "test/common/stats/stat_test_utility.h"
 #include "test/mocks/config/mocks.h"
+#include "test/mocks/event/mocks.h"
```
Comment (mattklein123) [inline test/common/secret/sds_api_test.cc:17]: Why is this needed in the revert? Seems unrelated?

Comment (yuval-k) [reply test/common/secret/sds_api_test.cc:17]: In my code I added it to mocks/grpc/mocks.h,so it was includes transitivly.. after the revert it is no longer in grpc.h

**Review by mattklein123 — APPROVED** [review]
Thanks!
